sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -242,6 +242,35 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
   }
 }
 
+object RapidsExecutorPlugin {
+  /**
+   * Return true if the expected cudf version is satisfied by the actual version found.
+   * The version is satisfied if the major and minor versions match exactly. If there is a requested
+   * patch version then the actual patch version must be greater than or equal.
+   * For example, version 7.1 is not satisfied by version 7.2, but version 7.1 is satisfied by
+   * version 7.1.1.
+   */
+  def cudfVersionSatisfied(expected: String, actual: String): Boolean = {
+    val expSplits = expected.split('.')
```

Comment:
nit:
```
val (expMajorMinor, expPatch) = expected.split('.').splitAt(2)
val (actMajorMinor, actPatch) = actual.split('.').splitAt(2)
if (expMajorMinor != actMajorMinor) {
   false
} else {
   ...
}
```

Reply:
This doesn't work as-is, but I get the idea.  I can do something similar in spirit.

Reply:
this seems to work
```scala
    val (expMajorMinor, expPatch) = expected.split('.').splitAt(2)
    val (actMajorMinor, actPatch) = actual.split('.').splitAt(2)
    actMajorMinor.startsWith(expMajorMinor) && {
      // if NaN make expected high, actual low so
      // that act is never greater than exp
      val expPatchInts = expPatch.map(v => Try(v.toInt).getOrElse(Int.MaxValue))
      val actPatchInts = actPatch.map(v => Try(v.toInt).getOrElse(Int.MinValue))
      val zipped = expPatchInts.zipAll(actPatchInts, 0, 0)
      zipped.forall { case (e, a) => e <= a }
    }
```

also takes care of the cases where expected patch is longer than the actual one:

```scala
    assert(!RapidsExecutorPlugin.cudfVersionSatisfied("7.0.1", "7.0"))
    assert(!RapidsExecutorPlugin.cudfVersionSatisfied("7.0.1.1", "7.0.1"))
    assert(!RapidsExecutorPlugin.cudfVersionSatisfied("7.0.1-special.1", "7.0.2"))
    assert(!RapidsExecutorPlugin.cudfVersionSatisfied("7.0.2.2.2", "7.0.2.2"))
```

Reply:
Thanks!  That looks cleaner. I updated to use something very similar, except I left the expected as a raw `toInt` since I'd rather know up-front that we have an unexpected string in the expected version rather than a more cryptic cudf versions don't match when they actually do.